script.js

```diff
@@ -1,4 +1,312 @@
+
+  // if player is added in between rounds,
+  // fill up player values history for previous rounds
+  // so that it does not appear as undefined
+  if (currentRound > 1) {
+    for (
+      var currentRoundCounter = 1;
+      currentRoundCounter < currentRound;
+      currentRoundCounter++
+    ) {
+      playerValuesHistory[playerValuesHistory.length - 1].push("n.a.");
+    }
+  }
+
+  generatePlayerTable();
+
+  if (playerNames.length > 1) {
+    document.getElementById("start-knockout-round-button").disabled = false;
+    document.getElementById("start-round-button").disabled = false;
+  }
+};
+
+var nameIsDuplicated = function (playerName) {
+  return playerNames.indexOf(playerName) !== -1;
+};
+
+var generatePlayerTable = function () {
+  var table = `<h5 class="text-center">--- Score Board ---</h5><div class="table-responsive"><table class="table table-bordered border-secondary text-center bg-white"><thead><tr><th scope="col">Players</th>`;
+
+  // generate player names
+  for (var counter = 0; counter < playerNames.length; counter++) {
+    table += `<th scope="col">${playerNames[counter]}</th>`;
+  }
+
+  table += `</tr></thead><tbody><tr><th>Score</th>`;
+
+  // generate player score
+  for (var counter = 0; counter < playerScores.length; counter++) {
+    table += `<td>${playerScores[counter]}</td>`;
+  }
+
+  table += `</tr><tr><th>Leader</th>`;
+
+  // generate leader
+  for (var counter = 0; counter < currentLeader.length; counter++) {
+    if (currentLeader[counter] == true) {
+      table += `<td>👑</td>`;
+    } else {
+      table += `<td></td>`;
+    }
+  }
+
+  table += `</tr><tbody></table></div>`;
+  document.getElementById("score-div").innerHTML = table;
+  document.getElementById("dice-div").style.display = "";
+};
+
+var toggleRunningKnockoutMode = function () {
+  if (runningKnockoutMode == "Running") {
+    runningKnockoutMode = "Knockout";
+    document.getElementById("running-history-div").style.display = "none";
+    if (currentKnockoutRound > 1) {
+      document.getElementById("knockout-history-div").style.display = "";
+    }
+    document.getElementById("start-round-button").style.display = "none";
+    document.getElementById("start-knockout-round-button").style.display = "";
+    document.getElementById("running-knockout-mode").innerHTML = "Knockout";
+    document.getElementById("running-knockout-mode").className =
+      "mode text-success";
+  } else {
+    runningKnockoutMode = "Running";
+    document.getElementById("knockout-history-div").style.display = "none";
+    if (currentRound > 1) {
+      document.getElementById("running-history-div").style.display = "";
+    }
+    document.getElementById("start-round-button").style.display = "";
+    document.getElementById("start-knockout-round-button").style.display =
+      "none";
+    document.getElementById("running-knockout-mode").innerHTML = "Running";
+    document.getElementById("running-knockout-mode").className =
+      "mode text-danger";
+  }
```

Comment:
Good job overall! You've exceeded the basic requirements by a whole lot so keep it up! one suggestion is to consistently document / write comments on your code